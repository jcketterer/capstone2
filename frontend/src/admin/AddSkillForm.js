import React, { useContext } from 'react';
import { Redirect } from 'react-router-dom';
import AdvocateForm from '../forms/AdvocateForm';
import UserContext from '../UserContext';
import AlertContext from '../AlertContext';
import AdvocateAPI from '../api';

const AddSkillForm = () => {
  const user = useContext(UserContext);

  const { setMessage } = useContext(AlertContext);

  const formFields = [
    {
      name: 'name',
      label: 'Name:',
      inputType: 'text',
      initialValue: '',
      placeholder: 'Please enter new skill name',
      required: true,
    },
  ];

  async function add(skill) {
    try {
      let newSkill = await AdvocateAPI.addSkill(skill);
      if (newSkill.name && newSkill.name === skill.name) {
        setMessage({
          text: `Skill ${newSkill.name} added successfully!`,
          variant: 'success',
        });
      }
    } catch (err) {
      setMessage({
        text: `Error Adding Skill ${skill.name}`,
        variant: 'danger',
      });
    }
  }

  if (!user) {
    return <Redirect to="/" />;
  }

  return (
    <div className="AddSkillForm">
      <div className="container">
        <AdvocateForm
          formTitle={`Add Skill`}
          fields={formFields}
          submitButtonText="Add Skill"
          processData={data => add(data)}
          backButtonTo={user.isAdmin ? '/admin' : '/skills'}
        />
      </div>
    </div>
  );
};

export default AddSkillForm;
