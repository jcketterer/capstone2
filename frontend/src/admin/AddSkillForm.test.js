import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AddSkillForm from './AddSkillForm';
import UserContext from '../UserContext';
import AlertContext from '../AlertContext.js';

it('should render without breaking', () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AlertContext.Provider value={{ setMessage: {} }}>
          <AddSkillForm />
        </AlertContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it('should match the snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ username: 'test' }}>
        <AlertContext.Provider value={{ setMessage: {} }}>
          <AddSkillForm />
        </AlertContext.Provider>
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
