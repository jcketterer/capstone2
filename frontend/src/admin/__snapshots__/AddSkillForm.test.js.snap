// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddSkillForm"
  >
    <div
      class="container"
    >
      <div
        class="AdvocateForm justify-content-center row"
      >
        <div
          class="col-md-8 col-sm-10"
        >
          <div
            class="card"
          >
            <h4
              class="card-header"
            >
              Add Skill
            </h4>
            <div
              class="card-body"
            >
              <form
                class=""
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="name"
                  >
                    Name:
                  </label>
                  <input
                    class="my-2 form-control"
                    id="name"
                    name="name"
                    placeholder="Please enter new skill name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <button
                  class="mr-2 btn btn-primary"
                  type="submit"
                >
                  Add Skill
                </button>
                <a
                  class="btn btn-secondary"
                  href="/skills"
                >
                  Back
                </a>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
